modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -190,10 +191,11 @@ object Inputs {
       extends OnDisk with SourceFile with AnyScalaFile with AnyScript {
     lazy val path: os.Path = base / subPath
   }
-  final case class ScalaFile(base: os.Path, subPath: os.SubPath)
+  case class ScalaFile(base: os.Path, subPath: os.SubPath)
```

Comment:
Inheriting from case classes is kind of an anti-pattern… They should be considered as effectively final. Better introduce a super-type if you really need `ScalaFile` and `ProjectSettings` to implement some common type in some contexts.

Reply:
Adding a method such as `def isProjectSettings: Boolean` to `ScalaFile` would be fine for example, if it can work for you here, rather than adding a new `ProjectSettings` class.